designer/server/src/main/scala/pl/touk/nussknacker/ui/config/DesignerConfig.scala

```diff
@@ -1,62 +1,242 @@
+
+    val fragmentPropertiesDocsUrl     = resolvedConfig.getAs[String]("fragmentPropertiesDocsUrl")
+    val repositoryGaugesCacheDuration = resolvedConfig.getDuration("repositoryGaugesCacheDuration")
+    val globalBuildInfo               = resolvedConfig.getAs[Map[String, String]]("globalBuildInfo")
+    val ssl                           = SslConfigParser.parseSslConfig(resolvedConfig)
+    val http                          = resolvedConfig.as[HttpConfig]("http")
+    val attachments                   = AttachmentsConfig.parse(resolvedConfig)
+
+    new DesignerConfig(
+      rawConfigWithUnresolvedVersion = rawConfig,
+      managersDir = managersDir,
+      configLoaderConfig = configLoaderConfig,
+      environment = environment,
+      usageStatisticsReportsConfig = usageStatisticsReportsConfig,
+      development = isDevelopmentMode,
+      metrics = metrics,
+      remoteEnvironment = remoteEnvironment,
+      counts = counts,
+      commentSettings = commentSettings,
+      deploymentCommentSettings = deploymentCommentSettings,
+      scenarioLabelConfig = scenarioLabelSettings,
+      scenarioStateTimeout = scenarioStateTimeout,
+      surveySettings = surveySettings,
+      tabs = tabs,
+      intervalTimeSettings = intervalTimeSettings,
+      environmentAlert = environmentAlert,
+      testDataSettings = testDataSettings,
+      enableConfigEndpoint = enableConfigEndpoint,
+      redirectAfterArchive = redirectAfterArchive,
+      componentDefinitionExtractionMode = componentDefinitionExtractionMode,
+      stickyNotesSettings = stickyNotesSettings,
+      deploymentsStatusesSynchronizationConfig = deploymentsStatusesSynchronizationConfig,
+      finishedDeploymentStatusesSynchronization = finishedDeploymentStatusesSynchronization,
+      componentLinks = componentLinks,
+      processToolbarConfig = processToolbarConfig,
+      questDbSettings = questDbSettings,
+      notifications = notifications,
+      fragmentPropertiesDocsUrl = fragmentPropertiesDocsUrl,
+      repositoryGaugesCacheDuration = repositoryGaugesCacheDuration,
+      globalBuildInfo = globalBuildInfo,
+      ssl = ssl,
+      http = http,
+      attachments = attachments,
+    )
+  }
+
+  private[ui] def parseComponentLinksConfig(resolvedConfig: Config) = {
+    implicit val optionListReader: ValueReader[List[ComponentLinkConfig]] = (config: Config, path: String) =>
+      ValueReader[List[Config]]
+        .read(config, path)
+        .map(_.as[ComponentLinkConfig])
+
+    resolvedConfig.getAs[List[ComponentLinkConfig]]("componentLinks").getOrElse(List.empty)
+  }
+
+  private def parseManagersDirs(rawConfig: Config): List[Path] = {
     val managersPath = "managersDirs"
-    if (rawConfig.resolved.hasPath(managersPath)) {
-      val managersDirs = rawConfig.resolved.getStringList(managersPath).asScala.toList
-      val paths        = managersDirs.map(_.convertToURL().toURI).map(Paths.get)
-      val invalidPaths = paths
```

Comment:
we don't need it?

Reply:
ok, I see it's moved

Reply:
Yeah. In some tests not related with configurability of Deployment Manager classpath  we create configuration with default `./managers` dir which doesn't exist and they blew up when I made this configurations checks eagerly processes in one place. TBH I think, this logic is closer to the service than to configuration parsing. Configuration parsing logic should be rather simple, we should check if path is in correct format. WDYT?